editor/sidebar/post-schedule/index.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import clickOutside from 'react-click-outside';
+import DatePicker from 'react-datepicker';
+import moment from 'moment';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from 'i18n';
+import { Component } from 'element';
+import { dateI18n, settings } from 'date';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import PostScheduleClock from './clock';
+import { getEditedPostAttribute } from '../../selectors';
+import { editPost } from '../../actions';
+
+class PostSchedule extends Component {
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			open: false,
+		};
+		this.toggleDialog = this.toggleDialog.bind( this );
+	}
+
+	toggleDialog( event ) {
+		event.preventDefault();
+		this.setState( { opened: ! this.state.opened } );
+	}
+
+	handleClickOutside() {
+		this.setState( { opened: false } );
+	}
+
+	render() {
+		const { date, onUpdateDate } = this.props;
+		const momentDate = date ? moment( date ) : moment();
+		const label = date
+			? dateI18n( settings.formats.datetime, date )
+			: __( 'Immediately' );
+		const handleChange = ( newDate ) => {
+			onUpdateDate( newDate.format( 'YYYY-MM-DDTHH:mm:ss' ) );
+		};
+		const is12HourTime = settings.formats.time.indexOf( 'a' ) !== -1 || settings.formats.time.indexOf( 'A' ) !== -1;
```

Comment:
Can be inaccurate because it's possible to [escape characters in the format](http://php.net/manual/en/function.date.php#refsect1-function.date-examples). In other words, `'H:i:s \a\w\e\s\o\m\e'` is not twelve hour time, but this logic would treat it as though it were.

Also, it could be simpler to force to lower/upper and test index of the one instance of 'a'/'A'.

Accounting for escaping could be tricky, particularly since JavaScript doesn't support lookbehind regular expression patterns. Dunno if there's a better way, but "flipping" the string and doing a look _ahead_ is one option:

```js
/a(?!\\)/i.test( 'h:i:s a'.split( '' ).reverse().join( '' ) )
// true
/a(?!\\)/i.test( 'h:i:s A'.split( '' ).reverse().join( '' ) )
// true
/a(?!\\)/i.test( 'H:i:s \\a\\w\\e\\s\\o\\m\\e'.split( '' ).reverse().join( '' ) )
// false
```